### PR TITLE
BOM-2949: Replace Travis with GitHub CI

### PR DESCRIPTION
**Pull request by UsamaSadiq**
**Issue:** [BOM-2949](https://openedx.atlassian.net/browse/BOM-2949)

### Description
- Replaced `Travis` with `GitHub CI`.
- Removed the script `scripts/build-stats-to-datadog.sh` since edX is no longer using `datadog` and the check was failing for quite some time.
- Removed `coveralls` in favor of using `codecov` for maintaining the code coverage reports.

### CODE REVIEW

**Review by ashultz0 — APPROVED** [review]
thanks!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - name: Run Tests
+      env:
+        TOXENV: ${{ matrix.toxenv }}
+      run: tox
+
+    - name: Run Coverage
+      if: matrix.toxenv=='py38'
+      uses: codecov/codecov-action@v2
+      with:
+        fail_ci_if_error: true
```
Comment (schenedx) [inline .github/workflows/ci.yml:43]: #NIT newline please.

**Review by schenedx — APPROVED** [review]
👍

File: .github/workflows/ci.yml (hunk 1 above)
Comment (UsamaSadiq) [reply .github/workflows/ci.yml]: Added new line at the end.

**Pull request merged**